knime_extension/src/nodes/spatialtool.py

```diff
@@ -1292,6 +1292,94 @@ def execute(self, exec_context: knext.ExecutionContext, input_table):
         return knut.to_table(grid, exec_context)
 
 
+############################################
+# Point to H3 Node
+############################################
+
+
+@knext.node(
+    name="Point to H3",
+    node_type=knext.NodeType.MANIPULATOR,
+    icon_path=__NODE_ICON_PATH + "PointToH3.png",
+    category=__category,
+    after="",
+)
+@knext.input_table(
+    name="Input Table",
+    description="Input table with the geometry",
+)
+@knext.output_table(
+    name="Output Table",
+    description="Output table with the H3 Cell Indices and hexagons column (optional).",
+)
+class PointToH3:
+    """Point to H3 node.
+    This node returns the H3 Cell Indices and hexagons (optional) for each point in the input table.
+    """
+
+    geo_col = knut.geo_col_parameter()
+
+    zoom = knext.IntParameter(
+        "Zoom level",
+        """The zoom level of the grid from 0 to 15 (default value is 8). The bigger the zoom level, the smaller the 
+        hexagon. If the zoom level is too small, the hexagon might be too big to fit in the input polygon which will
+        result in an error. A very small zoom level might result in a very large output table even for smaller 
+        input polygons. 
+        For more details about the zoom levels  refer to 
+        [Tables of Cell Statistics Across Resolutions.](https://h3geo.org/docs/core-library/restable/)
+        """,
+        default_value=8,
+        min_value=0,
+        max_value=15,
+    )
+
+    _COL_ID = "H3 Cell Index"
+    _COL_GEOMETRY = "geometry"
+
+    def configure(self, configure_context, input_schema):
+        self.geo_col = knut.column_exists_or_preset(
+            configure_context, self.geo_col, input_schema, knut.is_geo_point
+        )
+
```

Comment:
Use append on input_schema with the two new columns like here: https://github.com/spatial-data-lab/knime-geospatial-extension/blob/main/knime_extension/src/util/knime_utils.py#L680

Reply:
Ideally the node is not replacing the input table but simply appending the result columns to the existing input table

Reply:
If appending the columns is to much work we could also let the user choose an id column from the input table that gets appended to the result table